File: Algorithms/Implementation/Easy/Modified Kaprekar Numbers/c/kaprekar2/kaprekar.c
#include <math.h>
#include <stdio.h>
#include "kaprekar.h"

void Show(int *limits) {
	enum None {False, True};
	enum None none = True;

	for (int i = limits[0]; i <= limits[1]; i++) {
		int d = 0, n = i;
		while (n != 0) {
			d++;
			n /= 10;
		}
		long square = (long)pow(i, 2);
		int square_size = 0;
		long temp = square;
		while (temp != 0) {
			square_size++;
			temp /= 10;
		}
		int l = square_size > d ? square / (int)pow(10, d) : 0;
		int r = square - (square / (int)pow(10, d)) * (int)pow(10, d);

		if (l + r == i) {
			none = False;
			i == limits[1] ? printf("%d", i) : printf("%d ", i);
		}
	}

	if (none) {
		printf("INVALID RANGE");
	}
}

File: Algorithms/Implementation/Easy/Modified Kaprekar Numbers/python/kaprekar2/kaprekar.py
class KaprekarNumbers():
    def __init__(self, limits):
        self.__limits = limits
        self.__kaprekar_numbers = list()

    def __FindNumbers(self):
        for i in range(self.__limits[0], self.__limits[1] + 1):
            d = len(str(i))
            square = str(pow(i, 2))
            l = int(square[: len(square) - d]) if len(square) > d else 0
            r = int(square[len(square) - d:])

            if l + r == i:
                self.__kaprekar_numbers.append(i)

    def ShowNumbers(self):
        self.__FindNumbers()

        if len(self.__kaprekar_numbers) > 0:
            for k in self.__kaprekar_numbers:
                print(str(k) if k == self.__kaprekar_numbers[-1] else str(k) + ' ', end='')
        else:
            print("INVALID RANGE")